airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStoragePathFactory.kt

```diff
@@ -97,31 +137,31 @@ class ObjectStoragePathFactory(
             listOf(
                 PathVariable("NAMESPACE") { it.stream.descriptor.namespace ?: "" },
```

Comment:
I should add an explicit comment: if pattern is null, then the provider context is used.

Practically: `patternResolved = pattern ?: actualSubstitution`

Ie, we match literally against the actual name and namespace, but use patterns for everything else. (Because these operations happen by stream.)